File: motor3d 2.0.ConvertedToC#/Ordenamiento.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
namespace Motor3D.Utilidades
{
	public class Ordenamiento
	{
		public Ordenamiento() : base()
		{
		}

		public static void BubbleSort<Tipo>(ref Tipo[] Array) where Tipo : IComparable
		{
			Tipo inter = default(Tipo);
			for (int i = 0; i <= Array.GetUpperBound(0); i++) {
				for (int j = 0; j <= Array.GetUpperBound(0); j++) {
					if (Array[i].CompareTo(Array[j]) == 1) {
						inter = Array[i];
						Array[i] = Array[j];
						Array[j] = inter;
					}
				}
			}
		}

		public static void Sort<Tipo>(ref Tipo[] Array) where Tipo : IComparable<Tipo>
		{
			System.Array.Sort(Array);
		}
	}
}

File: motor3d 2.0/Ordenamiento.vb
Namespace Utilidades
    Public Class Ordenamiento
        Public Sub New()
            MyBase.New()
        End Sub

        Public Shared Sub BubbleSort(Of Tipo As IComparable)(ByRef Array() As Tipo)
            Dim inter As Tipo
            For i As Integer = 0 To Array.GetUpperBound(0)
                For j As Integer = 0 To Array.GetUpperBound(0)
                    If Array(i).CompareTo(Array(j)) = 1 Then
                        inter = Array(i)
                        Array(i) = Array(j)
                        Array(j) = inter
                    End If
                Next
            Next
        End Sub

        Public Shared Sub Sort(Of Tipo As IComparable(Of Tipo))(ByRef Array() As Tipo)
            System.Array.Sort(Array)
        End Sub
    End Class
End Namespace
